packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -35,6 +38,31 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
         this.chain = chain;
     }
 
+    async getSigner(type: SignerType) {
```

Comment:
we need to define the return type of this method such that if you provide `"ethers"`, typescript knows that you're returned an `ethers.Signer`, and if `"viem"`, the return type is `viem.WalletClient`

It'll look something like this:
```
type SignerMap = {
    ethers: ethers.Signer;
    viem: WalletClient;
};

async getSigner<Type extends SignerType = SignerType>(type: Type): Promise<SignerMap[Type]> {
```
but I'm having trouble nailing this down right now, have asked Maxwell for some help

Reply:
ah, for now we can do:
```
type SignerMap = {
    ethers: ethers.Signer;
    viem: WalletClient;
};

export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet> extends BaseWallet {
    private sessionKeySignerAddress?: string;
    chain: B;

    constructor(provider: ZeroDevEthersProvider<"ECDSA">, crossmintService: CrossmintService, chain: B) {
        super(provider, crossmintService);
        this.chain = chain;
    }

    async getSigner<Type extends SignerType = SignerType>(type: Type): Promise<SignerMap[Type]> {
        switch (type) {
            case "ethers":
                return this.provider.getSigner() as any;
            case "viem":
                const customeip1193Provider = new CustomEip1193Bridge(this.provider.getAccountSigner(), this.provider);
                const customTransport = custom({
                    async request({ method, params }) {
                        return customeip1193Provider.send(method, params);
                    },
                });
                const walletClient: viem.WalletClient = createWalletClient({
                    account: await this.getAddress(),
                    chain: getViemNetwork(this.chain),
                    transport: customTransport,
                }).extend(publicActions);
                return walletClient as any;
            default:
                logError("[GET_SIGNER] - ERROR", {
                    error: errorToJSON("Invalid signer type"),
                });
                throw new Error("Invalid signer type");
        }
    }